prowler/providers/aws/services/codepipeline/codepipeline_service.py

```diff
@@ -0,0 +1,148 @@
+        """Lists all CodePipeline pipelines in the specified region.
+
+        Retrieves all pipelines using pagination and creates Pipeline objects
+        for each pipeline found.
+
+        Args:
+            regional_client: AWS regional client for CodePipeline service.
+
+        Raises:
+            ClientError: If there is an AWS API error.
+        """
+        logger.info("CodePipeline - Listing pipelines...")
+        try:
+            list_pipelines_paginator = regional_client.get_paginator("list_pipelines")
+            for page in list_pipelines_paginator.paginate():
+                for pipeline in page["pipelines"]:
+                    pipeline_arn = f"arn:{self.audited_partition}:codepipeline:{regional_client.region}:{self.audited_account}:pipeline/{pipeline['name']}"
+                    if self.pipelines is None:
+                        self.pipelines = {}
+                    self.pipelines[pipeline_arn] = Pipeline(
+                        name=pipeline["name"],
+                        arn=pipeline_arn,
+                        region=regional_client.region,
+                    )
+        except ClientError as error:
+            if error.response["Error"]["Code"] == "AccessDenied":
+                logger.error(
+                    f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                )
+                if not self.pipelines:
+                    self.pipelines = None
+            else:
+                logger.error(
+                    f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _get_pipeline_state(self, pipeline):
+        """Retrieves the current state of a pipeline.
+
+        Gets detailed information about a pipeline including its source configuration
+        and tags.
+
+        Args:
+            pipeline: Pipeline object to retrieve state for.
+
+        Raises:
+            ClientError: If there is an AWS API error.
+        """
+        logger.info("CodePipeline - Getting pipeline state...")
+        try:
+            regional_client = self.regional_clients[pipeline.region]
+            pipeline_info = regional_client.get_pipeline(name=pipeline.name)
+            source_info = pipeline_info["pipeline"]["stages"][0]["actions"][0]
+            pipeline.source = Source(
+                type=source_info["actionTypeId"]["provider"],
+                location=source_info["configuration"].get("FullRepositoryId", ""),
+                configuration=source_info["configuration"],
+            )
+
+            # Get tags using list_tags_for_resource API
+            try:
+                tags_response = regional_client.list_tags_for_resource(
```

Comment:
Please, get the tags in a different function as in https://github.com/yyyy7246/prowler/blob/2ebff1bf24040930e33ce59b7d3a2bcfa1f9b68d/prowler/providers/aws/services/sns/sns_service.py#L64

Reply:
@yyyy7246 and a test for that function in the service test please.

Reply:
> @yyyy7246 and a test for that function in the service test please.

Hello @MrCloudSec,

I've refactored the code to move the tag retrieval logic into a separate `_list_tags_for_resource` function, following the pattern you pointed out in the SNS service. In addition, I've added a dedicated test for this function within the service tests. 

Thank you for the feedback—it helped me improve the code structure and testing approach. Please let me know if you have any further suggestions.